### PR TITLE
drop docker from makefile and README

### PR DESCRIPTION
**Pull request by MatousJobanek**
* rename `docker.mk` to `podman.mk` (contains only podman targets now)
* update the doc

### CODE REVIEW

File: make/docker.mk [hunk 1]
```diff
@@ -1,61 +0,0 @@
...
-	$(eval TARGET_REGISTRY:=docker.io)
-
-.PHONY: docker-push-to-os
-## Push the docker image to the OS internal registry
-docker-push-to-os: set-os-registry docker-image docker-push
-
-.PHONY: set-os-registry
-## Sets TARGET_REGISTRY:=$(shell oc get images.config.openshift.io/cluster  -o jsonpath={.status.externalRegistryHostnames[0]})
-set-os-registry:
-	$(eval TARGET_REGISTRY:=$(shell oc get images.config.openshift.io/cluster  -o jsonpath={.status.externalRegistryHostnames[0]}))
```
Comment (MatousJobanek) [inline make/docker.mk:57]: We could maybe start using the internal OpenShift registry to simplify the dev setup/environment.

Comment (MatousJobanek) [reply make/docker.mk:57]: I can take a look at it when I'll work on the CI/CD changes that will be needed for moving repos to kubesaw GH org, and for introducing standard semver releases.

**Review by alexeykazakov — APPROVED** [review]

**Comment by alexeykazakov** [comment]
infra issue

/retest

**Comment by MatousJobanek** [comment]
infra issue

/retest

**Review by xcoulon — APPROVED** [review]

**Review by rajivnathan — APPROVED** [review]

**Pull request merged**